packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -125,6 +168,77 @@ export abstract class SfdxCommandletExecutor<T>
     return;
   }
 
+  public outputResultPull(parser: ForcePullResultParser) {
+    const table = new Table();
+    const titleType = 'pull';
+
+    const successes: any = parser.getSuccesses();
+    const errors = parser.getErrors();
+    const pulledSource = successes ? successes.result.pulledSource : undefined;
+    if (pulledSource) {
+      const rows = pulledSource || (errors && errors.result);
+      const tableTitle = nls.localize(`table_title_${titleType}ed_source`);
+      const outputTable = this.getOutputTable(table, rows, tableTitle);
+
+      channelService.appendLine(outputTable);
+      if (pulledSource && pulledSource.length === 0) {
+        const noResults = nls.localize('table_no_results_found') + '\n';
+        channelService.appendLine(noResults);
+      }
+    }
+
+    if (errors) {
+      const { name, message, result } = errors;
+      if (result) {
+        const outputTable = this.getErrorTable(table, result, titleType);
+        channelService.appendLine(outputTable);
+      } else if (name && message) {
+        channelService.appendLine(`${name}: ${message}\n`);
+      }
+    }
+  }
+
+  protected getOutputTable(
+    table: Table,
+    rows: unknown,
+    outputTableTitle: string | undefined
+  ) {
+    const outputTable = table.createTable(
+      (rows as unknown) as Row[],
+      [
+        { key: 'state', label: nls.localize('table_header_state') },
+        { key: 'fullName', label: nls.localize('table_header_full_name') },
+        { key: 'type', label: nls.localize('table_header_type') },
+        { key: 'filePath', label: nls.localize('table_header_project_path') }
+      ],
+      outputTableTitle
+    );
+    return outputTable;
+  }
+
+  protected getErrorTable(table: Table, result: unknown, titleType: string) {
+    const outputTable = table.createTable(
+      (result as unknown) as Row[],
+      [
+        {
+          key: 'filePath',
+          label: nls.localize('table_header_project_path')
+        },
+        { key: 'error', label: nls.localize('table_header_errors') }
+      ],
+      nls.localize(`table_title_${titleType}_errors`)
+    );
+    return outputTable;
+  }
+
+  /**
+   * @description Base method (no-op) that is overridden by sub-classes
+   * forceSourcePush and forceSourcePull to update the local cache's
+   * timestamps post-operation, in order to be in sync for the
+   * "Detect Conflicts at Sync" setting.
+   */
+  protected updateCache(result: any): void {}
+
   public abstract build(data: T): Command;
```

Comment:
I love me some method docs.  I dunno if you need the @description annotation though

Reply:
![Screenshot 2023-05-19 at 12 09 08 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/76090802/27c052a4-d6ac-4910-8f80-00124cf3c6f7)
vs
![Screenshot 2023-05-19 at 12 09 19 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/76090802/f059336a-2243-49d3-a300-0a7593824e14)